### ISSUE TITLE
Transcode input audio to 16 kHz, 16-bit depth in the Python client

### ISSUE THREAD

**Issue by reuben** [mozilla/DeepSpeech#1022]
Both the C++ and the JS clients leverage SoX to do this, but the Python client doesn't. We should make them consistent.

**Comment by est31** [comment]
Note that it doesn't seem to work for the C++ client -- at least it didn't work on my input data that were not in that format. Maybe it is due to some sub-par downsampling algo used?

**Comment by pikumar** [comment]
This is the first time I am using deepspeech. Perhaps one wants to modify the deepspeech python executable to check for input type: (Only tested on python 3.6+ubuntu+python-magic installed)

```python
# -*- coding: utf-8 -*-
import re
import sys
import magic # Install python-magic for this. Tested on python 3.6

from deepspeech.client import main

if __name__ == '__main__':
    if len(sys.argv) < 3:
        print("Not enough arguments!")
        sys.exit(1)
    itype = magic.from_file(sys.argv[2])
    if "16 " not in itype:
        print("Error: Please encode your input file to 16000Hz")
        sys.exit(1)
    else:
        print("Input:", itype)
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

**Comment by lissyx** [comment]
@pikumar Thanks but:
 - that file is generated by pip on installation, we don't have control for that 
 - we already document the current limitation
 - that's still a workaround, we should rather fix that. Patches are welcome, if you have time :)

**Comment by pikumar** [comment]
@lissyx 
Can you please add the following line in native_client/python/client.py at line 69:
```python
assert fs == 16000, "Only 16000Hz input WAV files are supported for now!"
```

**Comment by lissyx** [comment]
@pikumar I'd be happy that you file an issue for this workaround and send a PR with that patch :)

**Comment by lissyx** [comment]
That was fixed in #1203

**Comment by lissyx** [comment]
I'll add test coverage and then we can close this.
